var/spack/repos/builtin/packages/openmc/package.py

```diff
@@ -43,15 +45,21 @@ class Openmc(CMakePackage):
     def cmake_args(self):
         options = ["-DCMAKE_INSTALL_LIBDIR=lib"]  # forcing bc sometimes goes to lib64
 
+        use_newer_options = self.spec.satisfies('@0.13.1:')
         if "+mpi" in self.spec:
             options += [
                 "-DCMAKE_C_COMPILER=%s" % self.spec["mpi"].mpicc,
                 "-DCMAKE_CXX_COMPILER=%s" % self.spec["mpi"].mpicxx,
             ]
+            if use_newer_options:
+                options += ["-DOPENMC_USE_MPI:BOOL=ON"]
 
-        options += [self.define_from_variant("openmp")]
-        options += [self.define_from_variant("optimize")]
-        options += [self.define_from_variant("debug")]
+        if use_newer_options:
+            options += [self.define_from_variant("OPENMC_USE_OPENMP", "openmp")]
+        else:
+            options += [self.define_from_variant("openmp")]
+            options += [self.define_from_variant("optimize")]
+            options += [self.define_from_variant("debug")]
 
         if "+optimize" in self.spec:
```

Comment:
This looks really hacky and will lead to non-deterministic builds: `openmp build_type=RelWithDebInfo +optimize` and `openmp build_type=Debug +optimize` both install the same thing but have different hashes. I would remove the `+optimize` and `+debug` variants entirely and rely on the existing `build_type` variant found in all CMake packages.